webapp/graphite/finders/__init__.py

```diff
@@ -42,37 +43,33 @@ def extract_variants(pattern):
 
 
 def match_entries(entries, pattern):
-    # First we check for pattern variants (ie. {foo,bar}baz = foobaz or barbaz)
-    matching = []
+  # First we check for pattern variants (ie. {foo,bar}baz = foobaz or barbaz)
+  matching = []
 
-    for variant in expand_braces(pattern):
-      matching.extend(fnmatch.filter(entries, variant))
+  for variant in expand_braces(pattern):
+    matching.extend(fnmatch.filter(entries, variant))
 
-    return list(_deduplicate(matching))
+  return list(_deduplicate(matching))
 
 
 """
-    Brace expanding patch for python3 borrowed from:
-    https://bugs.python.org/issue9584
+  Brace expanding patch for python3 borrowed from:
+  https://bugs.python.org/issue9584
 """
-def expand_braces(orig):
-    r = r'.*(\{.+?[^\\]\})'
-    p = re.compile(r)
-
-    s = orig[:]
-    res = list()
-
-    m = p.search(s)
-    if m is not None:
-      sub = m.group(1)
-      open_brace = s.find(sub)
-      close_brace = open_brace + len(sub) - 1
-      if sub.find(',') != -1:
-        for pat in sub.strip('{}').split(','):
-          res.extend(expand_braces(s[:open_brace] + pat + s[close_brace + 1:]))
-      else:
-          res.extend(expand_braces(s[:open_brace] + sub.replace('}', '\\}') + s[close_brace + 1:]))
+def expand_braces(s):
+  res = list()
+
+  m = EXPAND_BRACES_RE.search(s)
+  if m is not None:
+    sub = m.group(1)
+    open_brace = s.find(sub)
+    close_brace = open_brace + len(sub) - 1
```

Comment:
Well, you have the following below:
```
      for pat in sub.strip('{}').split(','):
        res.extend(expand_braces(s[:open_brace] + pat + s[close_brace + 1:]))
    else:
        res.extend(expand_braces(s[:open_brace] + sub.replace('}', '\\}') + s[close_brace + 1:]))
```

With `m.span(1)` you can drop the ` + 1` bit.
```
      for pat in sub.strip('{}').split(','):
        res.extend(expand_braces(s[:open_brace] + pat + s[close_brace:]))
    else:
        res.extend(expand_braces(s[:open_brace] + sub.replace('}', '\\}') + s[close_brace:]))
```
:-)